File: server/src/main/java/com/tomato/bookstore/model/Product.java
package com.tomato.bookstore.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

/** 商品实体类 */
@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "products")
public class Product {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String title;

  @Column(nullable = false)
  private BigDecimal price;

  @Column(nullable = false)
  private Integer rate;

  private String description;
  private String cover;
  private String detail;

  @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
  @Builder.Default
  @ToString.Exclude
  @EqualsAndHashCode.Exclude
  private List<Specification> specifications = new ArrayList<>();

  @OneToOne(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
  @ToString.Exclude
  @EqualsAndHashCode.Exclude
  @JsonIgnore
  private Stockpile stockpile;

  @Column(nullable = false)
  private LocalDateTime createdAt;

  private LocalDateTime updatedAt;
}

File: client/src/types/product.ts
/**
 * 商品规格接口
 */
export interface Specification {
  id?: number;
  item: string;
  value: string;
}

/**
 * 商品接口
 */
export interface Product {
  id?: number;
  title: string;
  price: number;
  rate: number;
  description?: string;
  cover?: string;
  detail?: string;
  specifications?: Specification[];
}

/**
 * 商品参数接口（用于创建/更新）
 */
export type ProductParams = Product;

/**
 * 商品库存接口
 */
export interface Stockpile {
  id?: number;
  amount: number;
  frozen: number;
  productId?: number;
}

/**
 * 商品库存参数接口（用于更新）
 */
export interface StockpileParams {
  amount: number;
  frozen: number;
}
